.bit/responses/2.5-Week 2 Step 5.md

```diff
@@ -1,44 +1,159 @@
+</details>
+<br>
+ 
+
+<details>
+<summary>❓How do I use Fetch?</summary>
+</br>
+We're calling the `fetch` function - notice the **await** keyword, which we need because `fetch` returns a **Promise**, which is a proxy for a value that isn't currently known. You can read about Javascript promises here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise. 
+
+Javascript is what we call a "synchronous" language, meaning operations in Javascript block other operations from executing until they are complete, creating a sense of single directional flow. This means that only one operation can happen at a time. However, in order to maximize efficiency (save time and resources), Javascript allows the use of asynchronous functions.
+</details>
+<br>
+
+<details>
+<summary>❓What is an async function?</summary>
+</br>
+Simply put, async functions allow other operations to continue running as they are being executed. Refer to this site for more information:https://dev.to/hardy613/asynchronous-vs-synchronous-programming-23ed.
+
+Promises are sychnronous objects, similar to their real life meaning, return a value at some point in the future, or a reason for why that value could not be returned - they represent the result of an async function that may or may not be resolved.
+
+> [Is JavaScript Synchronous or Asynchronous? What the Hell is a Promise?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+
+> [Master the JavaScript Interview: What is a Promise?](https://medium.com/better-programming/is-javascript-synchronous-or-asynchronous-what-the-hell-is-a-promise-7aa9dd8f3bfb)
+</details>
+<br>
+
+
+<details>
+<summary>❓What is the URL?</summary>
+</br>
+Notice that the URL is just the uriBase with the params we specified earlier appended on.
+
+For now, fill in the `method`  and `body`.  
+
+```js
+async function analyzeImage(img){
+    
+    const subscriptionKey = '<YOUR SUBSCRIPTION KEY>';
+    const uriBase = '<YOUR ENDPOINT>' + '/face/v1.0/detect';
+
+    let params = new URLSearchParams({
+        'returnFaceId': 'true',
+        'returnFaceAttributes': 'emotion'
+    })
+
+    
+    //COMPLETE THE CODE
+    let resp = await fetch(uriBase + '?' + params.toString(), {
+        method: '<METHOD>',  //WHAT TYPE OF REQUEST?
+        body: '<BODY>',  //WHAT ARE WE SENDING TO THE API?
+        headers: {
+            '<HEADER NAME>': '<HEADER VALUE>'  //do this in the next section
+        }
+    })
+
+    let data = await resp.json();
+    
+    return data; 
+}
+```
+</details>
+<br>
+
+
+<details>
+<summary>❓How do I specify Request Headers?</summary>
+</br>
+Go back to the Face API documentation here:https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236, and find the **Request headers** section. 

```

Comment:
Make sure to add a newline after `</br>` so that the markdown works! you be able to do `[here](the link)`